src/pages/ProfilePage.tsx

```diff
@@ -20,132 +19,113 @@ import PressableWithoutFocus from '@components/Pressable/PressableWithoutFocus';
+import * as ReportActions from '@userActions/Report';
+import * as SessionActions from '@userActions/Session';
 import CONST from '@src/CONST';
+import type {TranslationPaths} from '@src/languages/types';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
-import personalDetailsPropType from './personalDetailsPropType';
+import type SCREENS from '@src/SCREENS';
+import type {PersonalDetails, PersonalDetailsList, Report, Session} from '@src/types/onyx';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import type {EmptyObject} from '@src/types/utils/EmptyObject';
 
-const matchType = PropTypes.shape({
-    params: PropTypes.shape({
-        /** accountID passed via route /a/:accountID */
-        accountID: PropTypes.string,
+type ProfilePageOnyxProps = {
+    /** The personal details of the person who is logged in */
+    personalDetails: OnyxEntry<PersonalDetailsList>;
 
-        /** report ID passed */
-        reportID: PropTypes.string,
-    }),
-});
-
-const propTypes = {
-    /* Onyx Props */
-
-    /** The personal details of all users */
-    personalDetails: PropTypes.objectOf(personalDetailsPropType),
-
-    /** Route params */
-    route: matchType.isRequired,
+    /** The report currently being looked at */
+    report: OnyxEntry<Report>;
 
     /** Session info for the currently logged in user. */
-    session: PropTypes.shape({
-        /** Currently logged in user accountID */
-        accountID: PropTypes.number,
-    }),
-
-    ...withLocalizePropTypes,
+    session: OnyxEntry<Session>;
 };
 
-const defaultProps = {
-    // When opening someone else's profile (via deep link) before login, this is empty
-    personalDetails: {},
-    session: {
-        accountID: 0,
-    },
-};
+type ProfilePageProps = ProfilePageOnyxProps & StackScreenProps<ProfileNavigatorParamList, typeof SCREENS.PROFILE_ROOT>;
 
 /**
  * Gets the phone number to display for SMS logins
- *
- * @param {Object} details
- * @param {String} details.login
- * @param {String} details.displayName
- * @returns {String}
  */
-const getPhoneNumber = (details) => {
-    // If the user hasn't set a displayName, it is set to their phone number, so use that
-    const displayName = lodashGet(details, 'displayName', '');
+const getPhoneNumber = ({login = '', displayName = ''}: PersonalDetails | EmptyObject): string | undefined => {
+    // If the user hasn't set a displayName, it is set to their phone number
     const parsedPhoneNumber = parsePhoneNumber(displayName);
+
     if (parsedPhoneNumber.possible) {
-        return parsedPhoneNumber.number.e164;
+        return parsedPhoneNumber?.number?.e164;
     }
 
     // If the user has set a displayName, get the phone number from the SMS login
-    return details.login ? Str.removeSMSDomain(details.login) : '';
+    return login ? Str.removeSMSDomain(login) : '';
 };
 
-function ProfilePage(props) {
+function ProfilePage({personalDetails = {}, route, session, report}: ProfilePageProps) {
```

Comment:
```suggestion
function ProfilePage({personalDetails, route, session, report}: ProfilePageProps) {
```

I think we can remove this